Formula/git.rb

```diff
@@ -13,19 +13,25 @@ class Git < Formula
 
   option "with-blk-sha1", "Compile with the block-optimized SHA1 implementation"
   option "without-completions", "Disable bash/zsh completions from 'contrib' directory"
-  option "with-brewed-openssl", "Build with Homebrew OpenSSL instead of the system version"
-  option "with-brewed-curl", "Use Homebrew's version of cURL library"
-  option "with-brewed-svn", "Use Homebrew's version of SVN"
+  option "with-openssl", "Build with Homebrew's OpenSSL instead of using CommonCrypto"
+  option "with-curl", "Use Homebrew's version of cURL library"
+  option "with-subversion", "Use Homebrew's version of SVN"
+  option "with-perl", "Build against a custom Perl rather than system default"
   option "with-persistent-https", "Build git-remote-persistent-https from 'contrib' directory"
 
+  deprecated_option "with-brewed-openssl" => "with-openssl"
+  deprecated_option "with-brewed-curl" => "with-curl"
+  deprecated_option "with-brewed-svn" => "with-subversion"
+
   depends_on "pcre" => :optional
   depends_on "gettext" => :optional
-  depends_on "openssl" if build.with? "brewed-openssl"
-  depends_on "curl" if build.with? "brewed-curl"
+  depends_on "openssl" => :optional
+  depends_on "curl" => :optional
   depends_on "go" => :build if build.with? "persistent-https"
+  depends_on :perl => ["5.6", :optional]
   # Trigger an install of swig before subversion, as the "swig" doesn't get pulled in otherwise
   # See https://github.com/Homebrew/homebrew/issues/34554
-  if build.with? "brewed-svn"
+  if build.with? "subversion"
     depends_on "swig"
     depends_on "subversion" => "with-perl"
```

Comment:
> OK, then we should probably flip them on by default.

I have some mixed feelings about this to be honest, given Perl's general suckiness when it comes to backwards compatibility. It's likely that if you install `svn` from the bottle & consequently with the bindings built against Homebrew's `perl` you'll need to build `git` from source to ensure you're using the same `perl` throughout the tree.

I haven't checked since I left but I presume I can't access analytics anymore, so I'll take guidance from you here in terms of what the best course of action is. The downside obviously being that if `svn`'s `perl` option goes back to being `optional` then `git` needs to use the now banned `depends_on "subversion" => "with-perl"` dance.